File: leet-code/226. Invert Binary Tree/InvertBinaryTree226_2.java
public class InvertBinaryTree226_2 {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) return null;
        TreeNode newNode = new TreeNode(root.val);
        newNode.left = invertTree(root.right);
        newNode.right = invertTree(root.left);
        return newNode;
    }
}

File: leet-code/226. Invert Binary Tree/InvertBinaryTree226_2.kt
package com.example

class InvertBinaryTree226_2 {
    fun invertTree(root: TreeNode?): TreeNode? {
        if (root == null) {
            return null
        }
        val left = root.left
        root.left = invertTree(root.right)
        root.right = invertTree(left)
        return root
    }
}
